### PR TITLE
Make struct type into a record.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/70559.

Addresses boxing found in default equality on .netfx in this type.

### CODE REVIEW

File: src/Features/Core/Portable/DocumentIdSpan.cs [hunk 1]
```diff
@@ -7,28 +7,27 @@
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Text;
 
-namespace Microsoft.CodeAnalysis
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/DocumentIdSpan.cs:10]: view with whitespcae off.

**Review by akhera99 — APPROVED** [review]

File: src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpIntelliSense.cs [hunk 2]
```diff
@@ -36,7 +35,7 @@ public override async Task InitializeAsync()
             VisualStudio.Workspace.SetImportCompletionOption(false);
         }
 
-        [WpfTheory, CombinatorialData]
+        [WpfTheory(Skip = "https://github.com/dotnet/roslyn/issues/70582"), CombinatorialData]
```
Comment (ToddGrun) [inline src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpIntelliSense.cs:38]: >[WpfTheory(Skip = "https://github.com/dotnet/roslyn/issues/70582"), CombinatorialData] [](http://example.com/codeflow?start=8&length=86)

Is Gen not making a PR with his commit of this? ([https://github.com/genlu/roslyn/commit/492d12fe051dc8a07d0a982d191a484e892eedc1](https://github.com/genlu/roslyn/commit/492d12fe051dc8a07d0a982d191a484e892eedc1))

Comment (CyrusNajmabadi) [reply src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpIntelliSense.cs:38]: he is.  this is just unblocking whichever pr goes in first.

Comment (CyrusNajmabadi) [reply src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpIntelliSense.cs:38]: (this is a merge of his branch).

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**